Add recipe for dtb-mode

schspa opened the pull request:
### Brief summary of what the package does

dtb-mode is a major mode for viewing [device tree](https://www.devicetree.org/) dtb binary files.

### Direct link to the package repository

https://github.com/schspa/dtb-mode

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Quick first pass.  I think this looks like a minor mode, so I would recommend the `define-minor-mode` macro which will automatically fill in all of the minor-mode idioms like the lighter, the mode variable, the hook, the keymap (if any), the mode name, the docstring, etc.  A couple other items:

checkdoc (using version 0.6.2):
```
Warning (emacs): 
dtb-mode.el:45: All interactive functions should have documentation
```

- The `shell-command-to-string` call should run `shell-quote-argument` on `buffer-file-name`, since nothing stops `buffer-file-name` from having special characters like ``` ` ``` and `>` and so forth
- I wonder if it would make sense to make `dtb-mode-command` a defcustom, but that's not a blocker
- dtb-mode.el#L7: Prefer `;; Version` over `;; Package-Version` (MELPA automatically adds `Package-Version`)
- dtb-mode.el#L25: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))

schspa commented:
@riscy  All problems listed have been corrected. Please review it.

riscy commented on recipes/dtb-mode:
```diff
@@ -0,0 +1,3 @@
+(dtb-mode :repo "schspa/dtb-mode"
+          :fetcher github
+          :branch "main")
```
If this is your default branch, you needn't specify. :)
```suggestion
          :fetcher github)
```

schspa replied:
this branch setup has been removed.

riscy reviewed (commented):
Thanks, this looks good as a minor mode.  I think each one of these issues is alone very minor -- fixing them should be straightforward and then this looks like an easy merge.

- The call to `(read-only-mode)` is an on/off toggle, so depending on what state the buffer is in you'll either be turning read-only mode on or off, and it may not be what you intend.  `(read-only-mode +1)` would be unambiguous, but really I think your mode should leave the buffer's read-only state alone and leave that up to the user
- Turning the mode off calls `(erase-buffer)` before we even know if the file `(buffer-file-name)` even exists.  The user could (probably) always undo, but it would probably be safer to check if the file exists first
- Similarly it looks like turning the mode _on_ calls `(erase-buffer)` before we know whether the `dtc` executable exists.  You could potentially use a `let` block to try to call `dtc` -- in this case you could lead with a call to `executable-find` to ensure you're not erasing the buffer prematurely

schspa commented:
> Thanks, this looks good as a minor mode. I think each one of these issues is alone very minor -- fixing them should be straightforward and then this looks like an easy merge.
> 
> * The call to `(read-only-mode)` is an on/off toggle, so depending on what state the buffer is in you'll either be turning read-only mode on or off, and it may not be what you intend.  `(read-only-mode +1)` would be unambiguous, but really I think your mode should leave the buffer's read-only state alone and leave that up to the user

Thanks for the advice, and I have changed to `(read-only-mode +1)`, because of device tree blob is a binary file and shouldn't be modified by the user directly, I still need to set it to read-only by my minor mode.

> * Turning the mode off calls `(erase-buffer)` before we even know if the file `(buffer-file-name)` even exists.  The user could (probably) always undo, but it would probably be safer to check if the file exists first

Yes, fixed.

> * Similarly it looks like turning the mode _on_ calls `(erase-buffer)` before we know whether the `dtc` executable exists.  You could potentially use a `let` block to try to call `dtc` -- in this case you could lead with a call to `executable-find` to ensure you're not erasing the buffer prematurely

Fixed too.

riscy commented:
Looks good, thanks.  The `progn` surrounding your `if` on line 57 isn't strictly necessary, but that's not a blocker.

And, welcome to MELPA. :)
